### DISCUSSION TITLE
LVGL with a AZ-Touch Display from AZ-Delivery

### DISCUSSION THREAD

**Discussion by Performance20** [arendst/Tasmota#13349] [Q&A] [answered]
Hi,

I try to get run this display from AZ-Delivery. It is a 2,8" Spi TFT Display with Touch (XPT2046), TFT Driver is ILI9341. Resolution is 320x240.   Modul  ESP32-D0WDQ6 
I compiled the actual dev version (tasmota32-lvg), add the defines like in the docu (LVGL + Berry + Tasmota), add the example Berry script as autoexec.be and this display.ini.

:H,ILI9341,240,320,16,SPI,1,*,*,*,*,*,*,*,40
:I
EF,3,03,80,02
CF,3,00,C1,30
ED,4,64,03,12,81
E8,3,85,00,78
CB,5,39,2C,00,34,02
F7,1,20
EA,2,00,00
C0,1,23
C1,1,10
C5,2,3e,28
C7,1,86
36,1,48
37,1,00
3A,1,55
B1,2,00,18
B6,3,08,82,27
F2,1,00
26,1,01
E0,0F,0F,31,2B,0C,0E,08,4E,F1,37,07,10,03,0E,09,00
E1,0F,00,0E,14,03,11,07,31,C1,48,08,0F,0C,31,36,0F
11,80
29,80
:o,28
:O,29
:A,2A,2B,2C
:R,36
:0,48,00,00,00
:1,28,00,00,01
:2,88,00,00,02
:3,E8,00,00,03
:r,1
:TS,*
#

and configue the Pins:
{"NAME":"Display","GPIO":[6210,1,576,1,800,768,0,0,1,1,7264,992,0,0,736,672,1,1,1024,704,1,1,1,1,0,0,0,0,1,1,1,1,1,0,0,1],"FLAG":0,"BASE":1}

The screen comes up but something is writing dots and lines on the screen at random, the background is different than expected (green boxes that fade to black), touch does not work and the ESP32 reboots from time to time. If I delete Touch (:TS,*) in dispaly.ini, the lines on the screen are gone, Touch doesn't work either, but the ESP32 then seems to run stably. 

Here the Log without the (:TS,*) option:

00:00:00.002 HDW: ESP32-D0WDQ6 
00:00:00.018 UFS: FlashFS mounted with 296 kB free
00:00:00.047 CFG: Loaded from File, Count 27
00:00:00.077 CFG: Loaded from LKG File, Count 26
00:00:00.079 FRC: Some settings have been reset (2)
00:00:00.083 SPI: Hardware using GPIO18(CLK), GPIO23(MOSI) and GPIO19(MISO)
00:00:00.136 BRY: Berry initialized, RAM used=5518
00:00:00.146 BRY: no 'preinit.be'
00:00:00.167 DSP: File descriptor used
00:00:00.821 RSL: RESULT = {"POWER":"ON"}
00:00:00.822 RSL: POWER = ON
00:00:00.822 DSP: ILI9341!
00:00:01.352 Project tasmota - Tasmota Version 9.5.0.9(lvgl)-1_0_7_4(2021-10-12T14:02:30)
00:00:01.469 LVG: LVGL initialized
00:00:01.592 BRY: successfully loaded 'autoexec.be'
00:00:07.011 WIF: Connecting to AP1 Ratingen Channel 11 BSSId 78:8A:20:B4:C3:AB in mode 11n as tasmota-18C850-2128...
00:00:12.790 WIF: Connect failed with AP timeout
00:00:13.175 WIF: Connecting to AP1 Ratingen in mode 11n as tasmota-18C850-2128...
00:00:13.182 QPC: Reset
00:00:20.493 WIF: Connected
00:00:20.758 HTP: Web server active on tasmota-18C850-2128 with IP address 192.168.7.86
13:39:59.781 RSL: INFO1 = {"Info1":{"Module":"Display","Version":"9.5.0.9(lvgl)","FallbackTopic":"cmnd/DVES_18C850_fb/","GroupTopic":"cmnd/tasmotas/"}}
13:39:59.787 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-18C850-2128","IPAddress":"192.168.7.86"}}
13:39:59.807 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU"}}
13:39:59.810 RSL: RESULT = {"POWER":"OFF"}
13:39:59.811 RSL: POWER = OFF
13:40:03.819 RSL: STATE = {"Time":"2021-10-12T13:40:03","Uptime":"0T00:00:21","UptimeSec":21,"Heap":99,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":99,"MqttCount":0,"POWER":"OFF","Wifi":{"AP":1,"SSId":"Ratingen","BSSId":"F0:9F:C2:F1:EE:E2","Channel":6,"Mode":"11n","RSSI":48,"Signal":-76,"LinkCount":1,"Downtime":"0T00:00:15"}}
13:40:03.844 RSL: SENSOR = {"Time":"2021-10-12T13:40:03","ESP32":{"Temperature":53.3},"TempUnit":"C"}

with (:TS,*) option:

00:00:00.002 HDW: ESP32-D0WDQ6 
00:00:00.035 UFS: FlashFS mounted with 296 kB free
00:00:00.072 CFG: Loaded from File, Count 11
00:00:00.080 QPC: Count 1
00:00:00.084 SPI: Hardware using GPIO18(CLK), GPIO23(MOSI) and GPIO19(MISO)
00:00:00.149 BRY: Berry initialized, RAM used=5518
00:00:00.162 BRY: no 'preinit.be'
00:00:00.203 DSP: File descriptor used
00:00:00.213 TS: XPT2046
00:00:00.857 RSL: RESULT = {"POWER":"ON"}
00:00:00.859 RSL: POWER = ON
00:00:00.859 DSP: ILI9341!
00:00:01.390 Project tasmota - Tasmota Version 9.5.0.9(lvgl)-1_0_7_4(2021-10-12T14:50:33)
00:00:01.583 LVG: LVGL initialized
00:00:01.706 BRY: successfully loaded 'autoexec.be'
00:00:07.605 WIF: Connecting to AP1 Ratingen Channel 11 BSSId 78:8A:20:B4:C3:AB in mode 11n as tasmota-18C850-2128...
00:00:11.933 WIF: Connected
00:00:12.194 HTP: Web server active on tasmota-18C850-2128 with IP address 192.168.7.86
13:54:34.188 RSL: INFO1 = {"Info1":{"Module":"Display","Version":"9.5.0.9(lvgl)","FallbackTopic":"cmnd/DVES_18C850_fb/","GroupTopic":"cmnd/tasmotas/"}}
13:54:34.194 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-18C850-2128","IPAddress":"192.168.7.86"}}
13:54:34.204 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU"}}
13:54:34.210 RSL: RESULT = {"POWER":"OFF"}
13:54:34.213 RSL: POWER = OFF
13:54:34.227 QPC: Reset
13:54:38.233 RSL: STATE = {"Time":"2021-10-12T13:54:38","Uptime":"0T00:00:12","UptimeSec":12,"Heap":89,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":204,"MqttCount":0,"POWER":"OFF","Wifi":{"AP":1,"SSId":"Ratingen","BSSId":"78:8A:20:B4:C3:AB","Channel":11,"Mode":"11n","RSSI":76,"Signal":-62,"LinkCount":1,"Downtime":"0T00:00:06"}}
13:54:38.260 RSL: SENSOR = {"Time":"2021-10-12T13:54:38","ESP32":{"Temperature":53.3},"TempUnit":"C"}

here my addtional #defines:
#define USE_LVGL
#define USE_DISPLAY
#define USE_DISPLAY_LVGL_ONLY
#define USE_XPT2046
#define USE_UNIVERSAL_DISPLAY
#undef USE_DISPLAY_MODES1TO5
#undef USE_DISPLAY_LCD
#undef USE_DISPLAY_SSD1306
#undef USE_DISPLAY_MATRIX
#undef USE_DISPLAY_SEVENSEG

Any idea/suggestion what is wrong?

Helge

**Comment by Performance20** [comment]
In the meantime I added ":B,60,1" to the display.ini which gives be now the right background (blue background instead of green boxes) but the other behaviour remains the same.

Helge

**Reply by s-hadinger** [reply]
I don't have a SPI touchscreen, so I can't help you here. @gemu2015 wrote the driver and may be able to help you.

**Comment by gemu2015** [comment, marked as answer]
remove touch and use the default ili9341 descriptor from tasmota. (ILI9341_display.ini)
dont enable DMA at first.
probably dont use berry at first and #define SHOW_SPLASH
will give a display init message. now rotate display either by display rotate or :r,x in descriptor.
test display with displaytext commands
when display works add touch in descriptor.
you will also need the :M cmd in descriptor since XPT needs calibration.
look at ili9488 XPT example.
you will have to adapt the :R opcodes at least for your selected rotation to your hardware too.
it would be helpful to debug the touch coordinates to the console.
at least in scripter you may print them with
\>S
print %wtch(0)% - %wtch(1)% - %wtch(2)%

when this is done proceed with BERRY and LVGL

**Reply by Performance20** [reply]
Hi,

I have carried out the suggested steps.  Starting with only 9341 driver by compiling the  "FIRMWARE_DISPLAYS" with additional defines:
#define SHOW_SPLASH
#define USE_AWATCH
#undef USE_BERRY

Works fine by adding some commands via  console (e.g. displayrotate 3, displaymode 0, etc.) and editing pins as ILI9341...  pins (..CS,..DC).
I was able to show Text, Analog Watch as well as Json Values  +  a Touch Button ([b0:160:120:100:50:2:11:4:2:Rel 1:])Touch Button work 30%, means only 30% inside of the Button are responsive, Rest  is outside the button. This error is only valid in the x direction. Y values works fine. 
After I switched to the Universal Driver, use ILI9341 display.ini and add Touchscreen via  :TS command, display worked like expected but Touchscreen no longer work.

The  permanent crash of the boad seems to be solved by replace the EPS32 NoteMCU by another boad with different layout. Seems to be the board has design problems by accessing the spi bus and the attached RAM.

Question is now, why the Touch not working with Universal driver (I changed CS and DC to SPI CD, SPI DC in pin config to get running the display)?

**Reply by gemu2015** [reply]
your display model should show number 17 now.
in console you should see the init messages from ili9341 and xpt touch controller

first you must monitor the x and y coordinates of the touch screen.
i would suggest to enable scripter  #define USE_SCRIPT 
and paste this little script  into the scripter window and enable script.
\>D
\>S
print %wtch(0)% - %wtch(1)% - %wtch(2)%

now in console watch the touch status and coordinates.

don't use :M directive in display descriptor at first.

and set the touch rotation code for your desired rotation  to zero 00 (no touch rotation)

e.g.  for rotation 3
:3,E8,00,00,00

now with a stylus touch all 4 edges and write down the x,y values

then insert the :M
directive into descriptor and put in the x,y start and end values you got.
e.g
:M,340,3900,300,3800

now the console readings should be equivalent to your screen coordinates.

**Reply by gtortone** [reply]
Hi, @gemu2015 can you give more details on calibration task ?
I set to 00 my preferred rotation (1) but for each calibration touch I have 2 values (x,y)
Which is the correct touches sequence for calibration ? (I'm using top-left, bottom-left, top-right, bottom-right...)
At the end I have 8 values, which values I have to use to compose ':M' command ?
Thanks.

**Comment by gemu2015** [comment]
you should get 4 value pairs but 2 each should be more or less equal to each other.
the map command has to be set as follows:
your measured lower x value (from top.left and lower.left)
your measured upper x value (from top.right and lower.right)
your measured lower y value (from top.left and top.right)
your measured upper y value (from lower.left and lower.right)
the map directive will convert your digitizers min max to screen min max then

**Reply by gtortone** [reply]
ok, thanks ! I suspect that this calibration should be done in rotation=0 (native geometry).... true or false ?

**Reply by gemu2015** [reply]
to be honest i think so but since i do not have a resistive touch i can't prove.
i only have capacitive touch panels which i clearly prefer. (no calibration needed)

**Comment by Digl** [comment]
Hi!
I also have the AZ Touch Mod and would be interested in getting tasmota to run on it.
@Performance20 did you get this working correctly?
Could you share your binary/source?
BR, Matt

**Reply by Performance20** [reply]
No I had no time the lasts weeks in order to dig more in the details.
The version I used only partly worked. I used a very simple example (1 button). Sreen worked sometimes and and touch position was different to the botton on the sreen. Alsa the code was very unstable and reboots sometimes serveral times. I made some modification on the code caused by compile errors. In parallel I used an example from AZ (code lock example) and this worked like expected.
But as I mention, the last weeks I had no time for further investigation.

**Comment by s-hadinger** [comment]
I have done a prototype for a simple Lvgl application to calibrate touch screen super easily. It computes automatically the `:M` line. 

I will release very soon

**Comment by s-hadinger** [comment]
The Touch Screen calibration command is now released. See https://github.com/arendst/Tasmota/pull/14459

**Comment by Digl** [comment]
I got the display to show the example berry script.
I used the pre-combiled tasmota32-lvgl.bin used your pin settings:
{"NAME":"Display","GPIO":[6210,1,576,1,800,768,0,0,1,1,7264,992,0,0,736,672,1,1,1024,704,1,1,1,1,0,0,0,0,1,1,1,1,1,0,0,1],"FLAG":0,"BASE":1}
and the following display.ini:
```
:H,ILI9341,240,320,16,SPI,1,*,*,*,*,*,*,*,40
:S,2,1,1,0,40,20
:I
EF,3,03,80,02
CF,3,00,C1,30
ED,4,64,03,12,81
E8,3,85,00,78
CB,5,39,2C,00,34,02
F7,1,20
EA,2,00,00
C0,1,23
C1,1,10
C5,2,3e,28
C7,1,86
36,1,48
37,1,00
3A,1,55
B1,2,00,18
B6,3,08,82,27
F2,1,00
26,1,01
E0,0F,0F,31,2B,0C,0E,08,4E,F1,37,07,10,03,0E,09,00
E1,0F,00,0E,14,03,11,07,31,C1,48,08,0F,0C,31,36,0F
11,80
29,80
:o,28
:O,29
:A,2A,2B,2C
:R,36
:0,48,00,00,00
:1,28,00,00,01
:2,88,00,00,02
:3,E8,00,00,02
:M,340,3900,300,3800
#
```
then I used the following berry script:
```
#- start LVGL and init environment -#
lv.start()

hres = lv.get_hor_res()       # should be 320
vres = lv.get_ver_res()       # should be 240

scr = lv.scr_act()            # default screean object
f20 = lv.montserrat_font(20)  # load embedded Montserrat 20

#- Background with a gradient from black #000000 (bottom) to dark blue #0000A0 (top) -#
scr.set_style_bg_color(lv.color(0x0000A0), lv.PART_MAIN | lv.STATE_DEFAULT)
scr.set_style_bg_grad_color(lv.color(0x000000), lv.PART_MAIN | lv.STATE_DEFAULT)
scr.set_style_bg_grad_dir(lv.GRAD_DIR_VER, lv.PART_MAIN | lv.STATE_DEFAULT)

#- Upper state line -#
stat_line = lv.label(scr)
if f20 != nil stat_line.set_style_text_font(f20, lv.PART_MAIN | lv.STATE_DEFAULT) end
stat_line.set_long_mode(lv.LABEL_LONG_SCROLL)                                        # auto scrolling if text does not fit
stat_line.set_width(hres)
stat_line.set_align(lv.TEXT_ALIGN_LEFT)                                              # align text left
stat_line.set_style_bg_color(lv.color(0xD00000), lv.PART_MAIN | lv.STATE_DEFAULT)    # background #000088
stat_line.set_style_bg_opa(lv.OPA_COVER, lv.PART_MAIN | lv.STATE_DEFAULT)            # 100% background opacity
stat_line.set_style_text_color(lv.color(0xFFFFFF), lv.PART_MAIN | lv.STATE_DEFAULT)  # text color #FFFFFF
stat_line.set_text("Tasmota")
stat_line.refr_size()                                                                # new in LVGL8
stat_line.refr_pos()                                                                 # new in LVGL8

#- display wifi strength indicator icon (for professionals ;) -#
wifi_icon = lv_wifi_arcs_icon(stat_line)    # the widget takes care of positioning and driver stuff
clock_icon = lv_clock_icon(stat_line)

#- create a style for the buttons -#
btn_style = lv.style()
btn_style.set_radius(10)                        # radius of rounded corners
btn_style.set_bg_opa(lv.OPA_COVER)              # 100% backgrond opacity
if f20 != nil btn_style.set_text_font(f20) end  # set font to Montserrat 20
btn_style.set_bg_color(lv.color(0x1fa3ec))      # background color #1FA3EC (Tasmota Blue)
btn_style.set_border_color(lv.color(0x0000FF))  # border color #0000FF
btn_style.set_text_color(lv.color(0xFFFFFF))    # text color white #FFFFFF

#- create buttons -#
prev_btn = lv.btn(scr)                            # create button with main screen as parent
prev_btn.set_pos(20,vres-40)                      # position of button
prev_btn.set_size(80, 30)                         # size of button
prev_btn.add_style(btn_style, lv.PART_MAIN | lv.STATE_DEFAULT)   # style of button
prev_label = lv.label(prev_btn)                   # create a label as sub-object
prev_label.set_text("<")                          # set label text
prev_label.center()

next_btn = lv.btn(scr)                            # right button
next_btn.set_pos(220,vres-40)
next_btn.set_size(80, 30)
next_btn.add_style(btn_style, lv.PART_MAIN | lv.STATE_DEFAULT)
next_label = lv.label(next_btn)
next_label.set_text(">")
next_label.center()

home_btn = lv.btn(scr)                            # center button
home_btn.set_pos(120,vres-40)
home_btn.set_size(80, 30)
home_btn.add_style(btn_style, lv.PART_MAIN | lv.STATE_DEFAULT)
home_label = lv.label(home_btn)
home_label.set_text(lv.SYMBOL_OK)                 # set text as Home icon
home_label.center()

#- callback function when a button is pressed, react to EVENT_CLICKED event -#

def btn_clicked_cb(obj, event)
    var btn = "Unknown"
    if   obj == prev_btn  btn = "Prev"
    elif obj == next_btn  btn = "Next"
    elif obj == home_btn  btn = "Home"
    end
    print(btn, "button pressed")
end


prev_btn.add_event_cb(btn_clicked_cb, lv.EVENT_CLICKED, 0)
next_btn.add_event_cb(btn_clicked_cb, lv.EVENT_CLICKED, 0)
home_btn.add_event_cb(btn_clicked_cb, lv.EVENT_CLICKED, 0)
```

```
{"Display":{"Model":17,"Type":0,"Width":320,"Height":240,"Mode":0,"Dimmer":10,"Size":2,"Font":1,"Rotate":3,"Invert":0,"Refresh":2,"Cols":[16,8],"Rows":14}}
```

but the touch doesn't work:
![image](https://user-images.githubusercontent.com/4592352/149850435-5f239eb7-2c9f-410c-b2e2-f5a0debc20ec.png)

**Reply by s-hadinger** [reply]
You need to use the following `display.ini`:

```
:H,ILI9341,320,240,16,SPI,1,*,*,*,*,*,*,*,40
:S,2,1,1,0,40,20
:I
EF,3,03,80,02
CF,3,00,C1,30
ED,4,64,03,12,81
E8,3,85,00,78
CB,5,39,2C,00,34,02
F7,1,20
EA,2,00,00
C0,1,23
C1,1,10
C5,2,3e,28
C7,1,86
36,1,48
37,1,00
3A,1,55
B1,2,00,18
B6,3,08,82,27
F2,1,00
26,1,01
E0,0F,0F,31,2B,0C,0E,08,4E,F1,37,07,10,03,0E,09,00
E1,0F,00,0E,14,03,11,07,31,C1,48,08,0F,0C,31,36,0F
11,80
29,80
:o,28
:O,29
:A,2A,2B,2C
:R,36
:0,28,00,00,00
:1,88,00,00,01
:2,E8,00,00,02
:3,48,00,00,03
:i,20,21
:TS,*
:M,3946,401,241,3828
#
```

**Reply by beatagerger** [reply]
hallo, benötige auch Hilfe bezüglich Display.
Ich möchte es aber schwarz lassen, oder hellblau, nicht diese rote/blaue Konposition.
Mein Problem:
es gibt eine Binary : "arduitouch-esp32_ili9341_full_4MB_v0.6.2" was wirklich nach Installieren den Display korrekt einschaltet, jedoch ohne "buzze".
aber!!!!!!!!! 
Auf die Platte können noch weitere Elemente eingefügt werden, es gibt einige freie GPIO-s, aber diese kann ich nicht in Tasmota installieren...
Ich möchte jedoch nicht die Temperatur von Board, sondern von der angeschlossenen DHT11 sehen. Ich möchte auch Relais per Touch steuern können.
Diese Definition in Arduino IDE ist kein Problem (nur mit MQTT komme in IDE nicht klar) aber als Tasmota-Install will es irgendwie nicht...
Was mache ich falsch, dass es nicht funktioniert?

**Reply by s-hadinger** [reply]
In English please.

Trying to guess what you are asking for, I don't understand what "arduitouch-esp32_ili9341_full_4MB_v0.6.2" is but it's definitely not Tasmota. If your question is about openHASP, please refer to their web site

**Comment by beatagerger** [comment]
So let's get to grips with "English"



I don't understand where anyone is reading anything all over the world, probably outside of the EU, because here in the EU everyone (Microsoft, Google, Apple) had to commit themselves to changing the browser settings so that every consumer could read everything in their own system language.



So if someone wants to answer in the Google translator in English, they will still see their answer in their own language (in my case in German)



Why this still doesn't work on some systems, probably because of Penguin...



To issue:

HASwitch is a mini application from Tasmota...

in the "AZ TOUCH-MOD" module, the consumer is forced to install more chips, because although there are some slots free, no temperature sensors can be installed. Like HomPoint, the display only shows all "online devices/data.



If you install Tasmota "lcf", the GPIOs are not correct and only ILI can be installed without the touch display drivers (XPT2046 touch controller)

If you install "Tasmota Display Version" no sensors can be used again...



That's the problem...



I'm not curious about "board temperature", but would like to install DHT11 for inside and DS20b for outside temperature, so a thermostat...



But openHASP is not suitable for this, at least not the version as "AZ TOUCH_MOD", only the old version with Nextion


Gesendet von Mail<https://go.microsoft.com/fwlink/?LinkId=550986> für Windows

Von: ***@***.***>
Gesendet: Montag, 27. Juni 2022 10:41
An: ***@***.***>
Cc: ***@***.***>; ***@***.***>
Betreff: Re: [arendst/Tasmota] LVGL with a AZ-Touch Display from AZ-Delivery (Discussion #13349)


In English please.

Trying to guess what you are asking for, I don't understand what "arduitouch-esp32_ili9341_full_4MB_v0.6.2" is but it's definitely not Tasmota. If your question is about openHASP, please refer to their web site

**Reply by s-hadinger** [reply]
> HASwitch is a mini application from Tasmota...

No it's not. Where did you get this idea from?

> in the "AZ TOUCH-MOD" module, the consumer is forced to install more chips, because although there are some slots free, no temperature sensors can be installed. Like HomPoint, the display only shows all "online devices/data.

What are you trying to achieve?

> If you install "Tasmota Display Version" no sensors can be used again...

There is always the possibility of self-compile a version with the drivers you want

**Reply by beatagerger** [reply]
please, how??????

**Reply by beatagerger** [reply]
to reach:

thermostat

Sensors:
DHT11 (1x)
DS18b(1x)
Moinsure sensor (2x)
Relays (2x)

that was on mega with touch display shield, that should be exchanged (not a nice box)
Values ​​for Homeassistant (possibly also to switch relays, but you can use rules to automate, for watering flowers)

**Reply by sfromis** [reply]
To use such sensors in Tasmota, you need to configure GPIO pins to match your hardware. Many devices have a ready-made pin config in https://templates.blakadder.com/

**Reply by beatagerger** [reply]
do not understand???
I'm not that stupid either...
it's not the first Tasmota...

These 2 temperature sensors (by the way, they are also installed on the terrace and in the garden) are part of the minimum and insTasmota32.bin are also included...

Version sensors = no displays, only sensors...
Versions Display = no sensors, only display selection

Apparently I know the version description of Tasmoter better?

**Comment by beatagerger** [comment]
ich habe die "selber-Bau" probiert...
In Tasmota fehlt die  XPT2046!!!!!!!!!!
ILI hat keine richtige Pin´s!!!

das funktioniert(von openHASP) nur ohne Sensoren...

https://openhasp.com/0.6.3/displays/MSPxxxx/

![image](https://user-images.githubusercontent.com/58541216/175971753-82deb233-9fd7-4cf3-aeff-4f88e3817977.png)

**Reply by sfromis** [reply]
If you want to ask about openHASP, this is the wrong place. Tasmota is a very different project, but does include HASPmota which is "heavily inspired" by openHASP, when it comes to how to program it. But it is not compatible with openHASP, and how to create a setup is very different. I do not think that those files have anything to do With Tasmota. Better ask where it is relevant, not here.

For context, those files are from the latest release of openHASP.
https://github.com/HASwitchPlate/openHASP/releases

**Reply by beatagerger** [reply]
thanks for hint.
in Gitpod I tried to configure openHASP in "override" (GPIO´s for Tempsensensor) but it doesn't work.
Tried adding "USE DHT11" doesn't work either.
Tasmota in Gitpod the change works (you get a .bin) but the "touch driver" is missing
Pins are provided: 32=DHT11
33=DS18b20

but probably in Gitpod I'm not writing this in the right place...

the description is a bit confusing, and it is translated to system language, so it makes it even harder to understand...

how to get on gitpod is ok...
above "insect head" is "platformio overreid" what to choose...

**Reply by beatagerger** [reply]
> Wenn Sie nach openHASP fragen möchten, ist dies der falsche Ort. Tasmota ist ein ganz anderes Projekt, enthält aber HASPmota, das von openWASP "stark inspiriert" ist, wenn es darum geht, wie man es programmiert. Aber es ist nicht kompatibel mit openHASP, und wie man ein Setup erstellt, ist sehr unterschiedlich. Ich glaube nicht, dass diese Dateien etwas mit Tasmota zu tun haben. Fragen Sie besser, wo es relevant ist, nicht hier.
> 
> Für den Kontext stammen diese Dateien aus der neuesten Version von openHASP. https://github.com/HASwitchPlate/openHASP/releases

free GPIO:
25
26
A0
32
33
34
35
36
39

**Reply by beatagerger** [reply]
> Wenn Sie nach openHASP fragen möchten, ist dies der falsche Ort. Tasmota ist ein ganz anderes Projekt, enthält aber HASPmota, das von openWASP "stark inspiriert" ist, wenn es darum geht, wie man es programmiert. Aber es ist nicht kompatibel mit openHASP, und wie man ein Setup erstellt, ist sehr unterschiedlich. Ich glaube nicht, dass diese Dateien etwas mit Tasmota zu tun haben. Fragen Sie besser, wo es relevant ist, nicht hier.
> 
> Für den Kontext stammen diese Dateien aus der neuesten Version von openHASP. https://github.com/HASwitchPlate/openHASP/releases

i don't use openHAB
also not other "open products"
In Tasmota I have no idea what to install where.
Where does "override.h" go?
everything in config?
all page settings also in config?
in Arduino the display runs without problems, in Tasmota not.
No display of temperature, only from board....
The description on Tasmota page does not apply to Ili, but to other displays... I can upload, but it is displayed incorrectly and touch does not work either...
Touch works in openHASP, but my sensors don't...

**Reply by beatagerger** [reply]
> Wenn Sie nach openHASP fragen möchten, ist dies der falsche Ort. Tasmota ist ein ganz anderes Projekt, enthält aber HASPmota, das von openWASP "stark inspiriert" ist, wenn es darum geht, wie man es programmiert. Aber es ist nicht kompatibel mit openHASP, und wie man ein Setup erstellt, ist sehr unterschiedlich. Ich glaube nicht, dass diese Dateien etwas mit Tasmota zu tun haben. Fragen Sie besser, wo es relevant ist, nicht hier.
> 
> Für den Kontext stammen diese Dateien aus der neuesten Version von openHASP. https://github.com/HASwitchPlate/openHASP/releases
To be honest I don't really care if it's openHASP or HAStasmota, whatever...

I want to see my (all) sensor readings on the display....

I think stick with Nextion and Arduino...
NabuCasa brings everything to Google and Alexa...
On Nextion (also on a small NODEMCU ESP8266) I can also install with easyESP, the Nextion does the rest...

**Comment by beatagerger** [comment]
bitte nicht irre zu füren!!
wenn du etwas nicht kennst, heißt lange nicht, dass es die nicht gibt.....

ich habe den Kasten vor mir...

Habe alle 3 Variationen versucht...
Tasmota mit Display
Tasmota LVGL

und auch openHASP

Bitte, schreib genau...
Wegen Übersetzung (Browser) ist es zu verstehen, je schon schwer genug...

Was ist "openWASP"???

Wenn du "AZ-TOUCH-MOD" installierst, ist das erste, was du siehst "OPENHASP" in eine kleibe blaue Schild... Schöne, zarte blaue Farbe... Nicht wie von Tasmota, dunkelblau, als hätte man "7-Segment-Display"
das kommt, bevor das System überhaupt "hochfährt"

und hier die Varianten von ".bin"

http://ota.tasmota.com/tasmota32/

**Reply by beatagerger** [reply]
https://github.com/HASwitchPlate/openHASP

**Comment by beatagerger** [comment]
please don't be wrong!!
If you don't know something, it doesn't mean it doesn't exist.....

I have the box in front of me...

Tried all 3 variations...
Tasmota with display
Tasmota LVGL

and also openHASP

Please write carefully...
Due to translation (browser) it's hard enough to understand ever...

What is "openWASP"???

When you install "AZ-TOUCH-MOD" the first thing you see is "OPENHASP" in a sticky blue shield... Nice delicate blue color... Not like Tasmota, dark blue like having "7- segment display"
that comes before the system even "boots up"

and here the variants of ".bin"

**Comment by beatagerger** [comment]
PIN´s in Tasmota:

TFT_CS 5
TFT_DC 4
TFT_MOSI (in Tasmota-> "MOSI SPI1") 23
TFT_SCLK (in Tasmota -> "CLK SPI1") 18
TFT_RST (in Tasmota -> "Reset Display") 22
TFT_MISO (in Tasmota -> "MISO SPI1") 19
TFT_LED (in Tasmota -> "Display lihgt") 15

???
TOUCH_CS???????????
????
TOUCH_IRQ??????????

Pin´s in openHASP (this is okay)

https://openhasp.haswitchplate.com/0.6/displays/MSPxxxx/

or in Tasmota Ili shouldn't be used at all, but a completely different display, "MSP2807"????

**Reply by beatagerger** [reply]
> PIN´s in Tasmota:
> 
> TFT_CS 5 TFT_DC 4 TFT_MOSI (in Tasmota-> "MOSI SPI1") 23 TFT_SCLK (in Tasmota -> "CLK SPI1") 18 TFT_RST (in Tasmota -> "Reset Display") 22 TFT_MISO (in Tasmota -> "MISO SPI1") 19 TFT_LED (in Tasmota -> "Display lihgt") 15
> 
> ??? TOUCH_CS??????????? ???? TOUCH_IRQ??????????
> 
> Pin´s in openHASP (this is okay)
> 
> https://openhasp.haswitchplate.com/0.6/displays/MSPxxxx/
> 
> or in Tasmota Ili shouldn't be used at all, but a completely different display, "MSP2807"????

PIN´s in Tasmota:

TFT_CS 5
TFT_DC 4
TFT_MOSI (in Tasmota-> "MOSI SPI1") 23
TFT_SCLK (in Tasmota -> "CLK SPI1") 18
TFT_RST (in Tasmota -> "Reset Display") 22
TFT_MISO (in Tasmota -> "MISO SPI1") 19
TFT_LED (in Tasmota -> "Display lihgt") 15

???
TOUCH_CS???????????
????
TOUCH_IRQ??????????

Pin´s in openHASP (this is okay)

https://openhasp.haswitchplate.com/0.6/displays/MSPxxxx/

or in Tasmota Ili shouldn't be used at all, but a completely different display, "MSP2807"????

**Comment by s-hadinger** [comment]
This conversation is going nowhere. Please reopen a proper discussion if needed. 

Locking this discussion
